.github/workflows/tests.yml

```diff
@@ -0,0 +1,113 @@
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: ["3.9", "3.10", "3.11"]
+        # In order to save resources, only run particular
+        # matrix slots on other OS than Linux.
+        include:
+          - os: "macos-latest"
+            python-version: "3.11"
+          - os: "windows-latest"
+            python-version: "3.11"
+
+    env:
+      OS: ${{ matrix.os }}
+      PYTHON: ${{ matrix.python-version }}
+
+    defaults:
+      run:
+        shell: bash -el {0}
+
+    name: Python ${{ matrix.python-version }} on OS ${{ matrix.os }}
+    steps:
+
+    - name: Acquire sources
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 2
+
+    - name: CACHING - Anaconda packages
+      uses: actions/cache@v3
+      id: cache-pkg
+      with:
+        path: ~/conda_pkgs_dir
+        key:
+          conda-pkg-${{ runner.os }}-${{ runner.arch }}-py${{ matrix.python-version }}-${{
+          env.CACHE_NUMBER }}-${{ hashFiles('environment-test.yml') }}
+      env:
+        # Increase this value if `environment-test.yml` has not changed,
+        # but you still want to reset the cache.
+        CACHE_NUMBER: 0
+
+    - name: INSTALL - Anaconda setup (Mambaforge)
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        miniforge-variant: Mambaforge
+        miniforge-version: latest
+        mamba-version: "*"
+        use-mamba: true
+        channels: conda-forge,defaults
+        channel-priority: true
+        activate-environment: herbie-test
+        auto-activate-base: false
+        use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
+
+    - name: CACHING - Anaconda environment
+      uses: actions/cache@v3
+      id: cache-env
+      with:
+        path: ${{ env.CONDA }}/envs
+        key:
+          conda-env-${{ runner.os }}-${{ runner.arch }}-py${{ matrix.python-version }}-${{
+          env.CACHE_NUMBER }}-${{ hashFiles('environment-test.yml') }}
+      env:
+        # Increase this value if `environment-test.yml` has not changed,
+        # but you still want to reset the cache.
+        CACHE_NUMBER: 0
+
+    - name: DEBUG - Anaconda info
+      run: conda info
+    - name: DEBUG - Anaconda configuration
+      run: conda config --show
+    - name: DEBUG - Environment variables
+      run: printenv | sort
+    - name: DEBUG - Program paths
+      run: |
+        command -v conda
+        command -v mamba
+
+    - name: INSTALL - Update Anaconda environment
+      run:
+        mamba env update --name herbie-test --file environment-test.yml
+      if: steps.cache-env.outputs.cache-hit != 'true'
+
+    - name: INSTALL - Project
+      run: |
+        pip install --editable=.
+
+    - name: Run tests
+      env:
+        TMPDIR: ${{ runner.temp }}
+      run: |
+        pytest -vvv -k test_ecmwf
```

Comment:
Just running a single test here for the sake of saving resources while iterating on the baseline setup, and to see the full test matrix succeed faster. It think it can be adjusted within a subsequent iteration, after seeing the test suite succeed on your end.